docs/user-guide/lazy/execution.md

```diff
@@ -48,32 +48,3 @@ If your data requires more memory than you have available Polars may be able to
 {{code_block('user-guide/lazy/execution','stream',['scan_csv','collect'])}}
 
 We look at [streaming in more detail here](streaming.md).
-
-### Execution on a partial dataset
```

Comment:
Would it make sense to keep this section as a shorter paragraph suggesting that the user use `head` early on in the query to execute the query on a partial dataset?

Reply:
Would it not still run against the whole dataset, just at an earlier stage? Optionally we could advice to limit the the input data by limiting partitions or query somehow, but that does seems quite obvious.

Reply:
@ritchie46 yup, we get that.
We're just debating whether we should replace the suggestion of using `fetch` with a suggestion to use `head` in the beginning of the query to test queries on large datasets.

Reply:
Ah, right. That might be useful.

Reply:
I don't know if I am misunderstanding something about the idea or how `head` and `collect` work, but would it not run after the IO part of the query, which almost always is clearly the slowest part? If we add this instruction then we do need make sure that people don't expect too much from it...

Reply:
So I tried a dataset of mine (7'400'000 rows) and the code takes 3.040seconds to run, if I instead call:
```python
df = (pl
    .scan_parquet(...)
    .head(10_000)
```
I get back 9'303 rows (there is a filtering operation in my code, so it's expected that we lose some rows) in 0.044s.

Reply:
Ok, so I assume that for parquet Polars is able to "push down" this head while reading, but I am no so sure that applies to all file formats like CSV?

I added the user guide section back in a modified format so if you @npielawski or @rodrigogiraoserrao want to take a look?